app/pages/utils/system_operation_prerun.py

```diff
@@ -221,6 +222,74 @@ def get_buses_load_t_df(_pypsa_network, load_t_key):
 
     return resultant_df
 
+
+def get_marginal_costs(_pypsa_network, marginal_cost_key, carrier, country):
+
+    if carrier == "electricity":
+        consider_carriers = [
+            'AC', 'H2',
+            'battery', 'Li ion', 'residential rural heat',
+            'residential rural water tanks', 'services rural heat',
+            'services rural water tanks', 'residential urban decentral heat',
+            'residential urban decentral water tanks',
+            'services urban decentral heat',
+            'services urban decentral water tanks', 'urban central heat',
+            'urban central water tanks', 'solid biomass',
+            'low voltage', 'home battery'
+        ]
```

Comment:
Would be nice to keep the approach consistent and keep hardcoding in the config for definitions of such type.

Reply:
agree, done.

unfortunately, for gas and electricity we still need to keep a hard coded segment due to the fact that gas cannot be filtered by country...